### ISSUE TITLE
Output the number of files linted on a clean run for the stylish formatter 

### ISSUE THREAD

**Issue by jobara** [eslint/eslint#13725]
**The version of ESLint you are using.**
v7.10.0

**The problem you want to solve.**

In the default formatter (stylish) if all the files successfully pass linting there is no output. 
(see: https://github.com/eslint/eslint/blob/c1974b3f7169a8e5fab7007df92d02d8c1a8d5a3/lib/cli-engine/formatters/stylish.js#L100) However, this possess a potential issue where you may have misconfigured the file globbing and think that all of the files have passed.

**Your take on the correct solution to problem.**

While other formatters like the tap formatter provide more details, it may be enough to simply provide the total number of files linted after a successful run. This will still keep the information brief, but will help ensure that the intended files have been checked.

**Are you willing to submit a pull request to implement this change?**

I'm happy to submit a PR to address this. In fact the change can be as small as changing the [output](https://github.com/eslint/eslint/blob/c1974b3f7169a8e5fab7007df92d02d8c1a8d5a3/lib/cli-engine/formatters/stylish.js#L100) to:

```javascript
return total > 0 ? chalk.reset(output) : `${results.length} ${pluralize("file", results.length)} lint free`;
```

**Comment by anikethsaha** [comment]
Thanks for the issue. 

I support this, It can be a useful enhancement. 

In the meantime, you can use the `--debug` flag as well or even write your own formatter.

**Comment by anikethsaha** [comment]
Should this be done for other formatters that are present in this repo as well, WDYT?
like `codeframe`, `compact`, `table`, `tap` etc

**Comment by jobara** [comment]
> Should this be done for other formatters that are present in this repo as well, WDYT?
> like `codeframe`, `compact`, `table`, `tap` etc

I think so, although tap already reports the individual files linted so it may not be necessary for that one.
https://github.com/eslint/eslint/blob/c1974b3f7169a8e5fab7007df92d02d8c1a8d5a3/lib/cli-engine/formatters/tap.js#L85

I'm happy to work on this, if you would like the contribution. Please let me know.

**Comment by anikethsaha** [comment]
I support this change. 
We need support/thoughts from other members as well. 

Also, I think this has to go through [RFC](https://github.com/eslint/rfcs)

**Comment by mdjermanovic** [comment]
Changing formatter's output is a big breaking change for all tools that consume the output, and also an undesirable change for end-users who prefer the current behavior, since there is no opt-in.

I think the best option is to create a custom formatter based on the existing one.

**Comment by anikethsaha** [comment]
> Changing formatter's output is a big breaking change for all tools that consume the output,

agreed. 

Though I think it would be great to have this by default itself. I think this can wait for next major ? till then, a custom formatter can be used !

**Comment by mdjermanovic** [comment]
I think we shouldn't cause such a breakage of tools without a very strong reason, even in a major version, and we don't know for sure if most end-users would prefer that change.

It's easy to make a custom formatter out of an existing one, and there are also available formatters on npm. For instance, [multiple](https://www.npmjs.com/package/eslint-formatter-multiple) can be configured to show output from both the `stylish` and [summary](https://www.npmjs.com/package/eslint-summary) formatters.

**Comment by nzakas** [comment]
Agree with @mdjermanovic. We can’t really change the stylish formatter that much especially because it is the default formatter and was intentionally designed to not output anything on success.

**Comment by nzakas** [comment]
The best option here is to create a custom formatter, so closing.

**Comment by jobara** [comment]
Thank you for the review, consideration and discussion, as well as the suggestions for alternatives.
